### PR TITLE
Only mention “public” in post status option if relevant.

### PR DESCRIPTION
**Pull request by rvdsteege**
I found the post status setting options a bit overwhelming:

Scherm­afbeelding 2024-12-04 om 15 28 36

This PR changes the option to only denote "public" if that is the case for the post status:

Scherm­afbeelding 2024-12-04 om 15 36 45

### CODE REVIEW

**Comment by remcotolsma** [comment]
I have struggled with this notation for a while. The disadvantage of this proposal is that if there are only `'public' => false` post statuses, so no public post statuses at all, the visibility remains a guess for users. I have also tried to shorten this with emojis or something, but that was not necessarily very clear either. That is how I eventually came up with this notation, it is not perfect, but I doubt whether this PR is an improvement. The backtics are also more of a markdown thing, should we use it like that? We need custom selects → https://developer.chrome.com/blog/rfc-customizable-select, https://open-ui.org/components/customizableselect/ 😏.

**Review by remcotolsma — COMMENTED** [review]
I have struggled with this notation for a while. The disadvantage of this proposal is that if there are only `'public' => false` post statuses, so no public post statuses at all, the visibility remains a guess for users. I have also tried to shorten this with emojis or something, but that was not necessarily very clear either. That is how I eventually came up with this notation, it is not perfect, but I doubt whether this PR is an improvement. The backtics are also more of a markdown thing, should we use it like that? We need custom selects → https://developer.chrome.com/blog/rfc-customizable-select, https://open-ui.org/components/customizableselect/ 😏.

File: admin/settings-field-post-types-support.php [hunk 1]
```diff
@@ -93,40 +93,64 @@ function ( $post_status_object ) {
...
-							printf(
-								'<option value="%s" %s>%s</option>',
-								\esc_attr( $value ),
-								\selected( $value, $current, false ),
-								\esc_html( $label )
-							);
+						foreach ( $groups as $group ) {
+							if ( 0 === count( $group['options'] ) ) {
+								continue;
+							}
```
Comment (remcotolsma) [inline admin/settings-field-post-types-support.php:132]: I prefer to use https://www.php.net/array_filter so that there is no such logic in the foreach.

**Review by remcotolsma — CHANGES_REQUESTED** [review]
Nice solution, smart to use `<optgroup>` elements for this.

File: admin/settings-field-post-types-support.php (hunk 1 above)
Comment (rvdsteege) [reply admin/settings-field-post-types-support.php]: Changed in https://github.com/pronamic/pronamic-post-expiration/pull/3/commits/85d7052d0de94d9be7114aebc5e05ed5ada2e155.

**Review by remcotolsma — APPROVED** [review]

**Pull request merged**